schema/handler_test.go

```diff
@@ -98,31 +97,138 @@ func TestHandler(t *testing.T) {
 	t.Run("case=get default schema", func(t *testing.T) {
 		server := getFromTS(config.DefaultIdentityTraitsSchemaID, http.StatusOK)
 		file := getFromFS(config.DefaultIdentityTraitsSchemaID)
-		require.Equal(t, file, server)
+		require.JSONEq(t, string(file), string(server))
 	})
 
 	t.Run("case=get other schema", func(t *testing.T) {
 		server := getFromTS("identity2", http.StatusOK)
 		file := getFromFS("identity2")
-		require.Equal(t, file, server)
+		require.JSONEq(t, string(file), string(server))
 	})
 
 	t.Run("case=get unreachable schema", func(t *testing.T) {
 		reason := getFromTS("unreachable", http.StatusInternalServerError)
-		require.Contains(t, reason, "could not be found or opened")
+		require.Contains(t, string(reason), "could not be found or opened")
 	})
 
 	t.Run("case=get no-file schema", func(t *testing.T) {
 		reason := getFromTS("no-file", http.StatusInternalServerError)
-		require.Contains(t, reason, "could not be found or opened")
+		require.Contains(t, string(reason), "could not be found or opened")
 	})
 
 	t.Run("case=get directory schema", func(t *testing.T) {
 		reason := getFromTS("directory", http.StatusInternalServerError)
-		require.Contains(t, reason, "could not be found or opened")
+		require.Contains(t, string(reason), "could not be found or opened")
 	})
 
 	t.Run("case=get not-existing schema", func(t *testing.T) {
 		_ = getFromTS("not-existing", http.StatusNotFound)
 	})
 }
+
+func TestHandlerAll(t *testing.T) {
+	conf, reg := internal.NewFastRegistryWithMocks(t)
+	router := x.NewRouterPublic()
+	reg.SchemaHandler().RegisterPublicRoutes(router)
+	ts := httptest.NewServer(router)
+	defer ts.Close()
+
+	schemas := schema.Schemas{
+		{
+			ID:     "default",
+			URL:    urlx.ParseOrPanic("file://./stub/identity.schema.json"),
+			RawURL: "file://./stub/identity.schema.json",
+		},
+		{
+			ID:     "identity2",
+			URL:    urlx.ParseOrPanic("file://./stub/identity-2.schema.json"),
+			RawURL: "file://./stub/identity-2.schema.json",
+		},
+	}
+
+	getSchemaById := func(id string) *schema.Schema {
+		s, err := schemas.GetByID(id)
+		require.NoError(t, err)
+		return s
+	}
+
+	getFromTS := func(expectCode, page, perPage int) []byte {
+		res, err := ts.Client().Get(fmt.Sprintf("%s/schemas?page=%d&per_page=%d", ts.URL, page, perPage))
+		require.NoError(t, err)
+		body, err := ioutil.ReadAll(res.Body)
+		require.NoError(t, err)
+		require.NoError(t, res.Body.Close())
+
+		require.EqualValues(t, expectCode, res.StatusCode, "%s", body)
+		return body
+	}
+
+	getFromFS := func(id string) []byte {
+		raw, err := os.ReadFile(strings.TrimPrefix(getSchemaById(id).RawURL, "file://"))
+		require.NoError(t, err)
+		return raw
+	}
```

Comment:
Actually, I would prefer using `t.Run` so you can even share the registry and stuff, makes it a bit faster.

Reply:
Which reminds me: in case one of the schemas is not found, should the list API omit it or return 404? Not sure what would be the best approach here.

Reply:
Yeah the problem was the api only returns the schemas that actually exist, so the test can't compare it to the list where also non existent schemas are listed

Reply:
But IRL it might happen that because of misconfiguration some old, unused schema is still around that is not resolvable. That's why I am bringing up the question how the API should behave :wink: 
You can also set config values mutliple times btw, it is hot reloaded. So you can set different schemas for each subtest.